sockeye/inference.py

```diff
@@ -1242,11 +1270,12 @@ def _encode(self, sources: mx.nd.NDArray, source_length: int) -> List[ModelState
         """
         Returns a ModelState for each model representing the state of the model after encoding the source.
 
-        :param sources: Source ids. Shape: (batch_size, bucket_key, num_factors).
+        :param sources: Source ids. Shape: (batch_size, num_input_langs, bucket_key, num_factors).
         :param source_length: Bucket key.
         :return: List of ModelStates.
         """
-        return [model.run_encoder(sources, source_length) for model in self.models]
+
+        return [model.run_encoder(sources[:, i], source_length) for i, model in enumerate(self.models)]
```

Comment:
this wouldn't work for single source ensemble models as sources would just contain the data for a single language, would it?